Vagrantfile

```diff
@@ -39,10 +39,13 @@ end
 # and ansible_local (see https://github.com/fgrehm/vagrant-lxc/issues/398)
 # once this issue is resolved, this can be probably removed.
 def get_provider
-  if ARGV[1] and (ARGV[1].split('=')[0] == "--provider" or ARGV[2])
-    return (ARGV[1].split('=')[1] || ARGV[2])
-  end
+  provider_index = ARGV.index('--provider')
+  if (provider_index && ARGV[provider_index + 1])
+     return ARGV[2]
 
+  elsif ARGV.index('--provider=lxc')
+     return "lxc"
```

Comment:
If I wanted to be picky I'd say we could explode the string on `=` to get the provider, but since the only goal of this method is to know if it is lxc or not, don' feel obligated ^^